### ISSUE TITLE
gpinitsystem can get into an infinite recursion.

### ISSUE THREAD

**Issue by berlin-ab** [greenplum-db/gpdb#7160]
TL; DR gpinitsystem -c gpinitsystem caused the shell script to source itself in an infinite recursion but the logging was catting my intended config file. 😄

This pr avoids issues when the config files are on the path.

Ex: I was following the Greenplum oss blog post for Ubuntu and used a copy of $GPHOME/docs/cli_help/gpconfigs/gpinitsystem_singlenode in my working directory to initialize my database. Later I decided to make it clustered and removed the _singlenode, thus winding up with a situation where the shell script sourced itself.

See this closed PR for a suggested change, and why it was not merged:

https://github.com/greenplum-db/gpdb/pull/6495
